app/src/main/java/com/vrem/wifianalyzer/wifi/collectdata/CollectDataFragment.java

```diff
@@ -0,0 +1,142 @@
+    private EditText currentLocationEt;
+    private static String objectString = "";
+    private static JSONObject jsonObject=new JSONObject();
+
+
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.collect_data, container, false);
+        MainContext.INSTANCE.getScannerService().register(this::update);
+
+        saveBtn = view.findViewById(R.id.saveCollectedData);
+        currentLocationEt = view.findViewById(R.id.collectDataEt);
+
+        // loop through wifiDetails and put() all bssid and current location
+        // call writeJSONtoFile with !!jsonObject!! IFF we really added any new values
+
+        saveBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                if (currentLocationEt.getText().toString().isEmpty())
+                    try {
+                        Log.d(TAG, "onClick: " + WiFiData.wiFiDetails.get(0).getBSSID()+" "+WiFiData.wiFiDetails.get(0).getTitle().split(" ")[0]);
+                        jsonObject.put(WiFiData.wiFiDetails.get(0).getBSSID(),WiFiData.wiFiDetails.get(0).getTitle().split(" ")[0]);
+                        writeJSONtoFile(MainContext.INSTANCE.getContext(), currentLocationEt.getText().toString(), jsonObject);
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+            }
+        });
+
+       /* readJSON(currentLocationEt.toString());*/
+        return view;
+    }
+
+
+    @Override
+    public void update(@NonNull WiFiData wiFiData) {
+
+        Settings settings = MainContext.INSTANCE.getSettings();
+        WiFiBand wiFiBand = settings.getWiFiBand();
+        Predicate<WiFiDetail> predicate = new WiFiBandPredicate(wiFiBand);
+        WiFiData.wiFiDetails = wiFiData.getWiFiDetails(predicate, SortBy.STRENGTH);
+    }
+
+    private String readJSON(String filename) {
+        File directory = Environment.getExternalStorageDirectory();
+        File file = new File(directory + "/Notes/", filename);
+
+        StringBuilder text = new StringBuilder();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            String line;
+
+            while ((line = br.readLine()) != null) {
+
+                text.append(line);
+                text.append('\n');
+            }
+            JSONObject jsonObject = new JSONObject(String.valueOf(text));
+            for (int i = 0; i<jsonObject.length(); i++) {
+                if (!objectString.contains(jsonObject.getString((String) jsonObject.names().get(i)))) {
+                    objectString = objectString + " " + jsonObject.getString((String) jsonObject.names().get(i)) + "\n";
+                }
+            }
+            /* Toast.makeText(mContext, objectString, Toast.LENGTH_SHORT).show();*/
+            br.close();
+        } catch (IOException e) {
```

Comment:
at least a printStackTrace here? 

Ideally should wrap and rethrow in a custom exception so the UI can notify the frontend